src/chat.rs

```diff
@@ -3435,7 +3426,27 @@ pub async fn get_chat_contacts(context: &Context, chat_id: ChatId) -> Result<Vec
                FROM chats_contacts cc
                LEFT JOIN contacts c
                       ON c.id=cc.contact_id
-              WHERE cc.chat_id=?
+              WHERE cc.chat_id=? AND cc.add_timestamp >= cc.remove_timestamp
+              ORDER BY c.id=1, c.last_seen DESC, c.id DESC;",
+            (chat_id,),
+            |row| row.get::<_, ContactId>(0),
+            |ids| ids.collect::<Result<Vec<_>, _>>().map_err(Into::into),
+        )
+        .await?;
+
+    Ok(list)
+}
+
+/// Returns a vector of contact IDs for given chat ID that are no longer part of the group.
+pub async fn get_past_chat_contacts(context: &Context, chat_id: ChatId) -> Result<Vec<ContactId>> {
+    let list = context
+        .sql
+        .query_map(
+            "SELECT cc.contact_id
+               FROM chats_contacts cc
+               LEFT JOIN contacts c
+                      ON c.id=cc.contact_id
+              WHERE cc.chat_id=? AND cc.add_timestamp < cc.remove_timestamp
               ORDER BY c.id=1, c.last_seen DESC, c.id DESC;",
```

Comment:
Maybe better sort by `remove_timestamp`? But then we shouldn't bump it unnecessarily in other places

Reply:
True, but we shouldn't bump `remove_timestamp` just so, otherwise we need to wait for 60 days (timestamp expiration period) so that the sorting is correct if we decide to sort by `remove_timestamp`. Reminding about this just in case

Reply:
> Where do we bump `remove_timestamp` unnecessarily?

It seems that only in `update_chat_contacts_table()`, but AFAIU this function is only for compatibility with old Delta Chat and MUA, so maybe that's not critical.